From f40817f470e4b38db6566e2ef70b362f41544bdb Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 29 Sep 2020 22:39:47 +0200
Subject: [PATCH] Some devices have proximity sensor reporting NaN as max range
 for some reason. Make them behave standard way by setting 5 cm

Change-Id: I3c39e3e914a05903c140235702e0480d2d58a612

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index 8aa68d58f41..6d5c8181b03 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -570,7 +570,7 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
             if (mProximitySensor != null) {
                 mProximityThreshold = Math.min(mProximitySensor.getMaximumRange(),
                         TYPICAL_PROXIMITY_THRESHOLD);
-		        if (Float.isNaN(mProximityThreshold)) {
+                if(Float.isNaN(mProximityThreshold)) {
                     mProximityThreshold = 5.0f;
                 }
             }
-- 
2.25.1

